### ISSUE TITLE
Comments in config.xml get overwritten by mod

### ISSUE THREAD

**Issue by rkagerer** [KSPSnark/IndicatorLights#3]
I witnessed XML comments manually added to my config.xml repeatedly getting wiped out.  First I thought I was doing something wrong (e.g. making changes while KSP was running) but I tried a test where I stopped the game, added comments, and then ran the game, and saw them get wiped out.

This isn't a major issue but it might be nice if there were a way to maintain the fidelity of the configuration file.  For what it's worth I'm pretty sure I've seen other mods using json-like formats that don't have this problem.

Maybe the Configuration class could be modified to only save the file if it didn't exist in the first place.

**Comment by KSPSnark** [comment]
Yep.  The issue is that, like most mods, IndicatorLights follows the path-of-least-resistance and uses the PluginConfiguration class to handle its configuration.  That class isn't super friendly to the use case you present-- it's not nicely equipped to deal with "did this exist", "did it not exist", "only write what you need to instead of overwriting the whole file", tracking whether things changed or not with a "dirty" flag, etc.  It's possible to do _some_ working around in code... but not convenient, and basically involves writing kludge code to try to reverse-engineer and outsmart PluginConfiguration.

And even then, it's still not going to be ideal.  For example:  Suppose the file exists, but there's a missing value?  It needs to be inserted.  But the only code in PluginConfiguration for writing _anything_ to the file involves overwriting _everything_ in the file, which means any comments or anything else would get wiped out.

Basically, what you want ("the settings file should be considered user-generated content, and should never wipe out what the user edited") can't happen, as long as PluginConfiguration is what I'm using; and I'm not about to go and author my own custom config reading/writing solution.  Any changes I make to try to "optimize" this would, IMHO, make it worse-- instead of quickly wiping out anything you do, it would lull you into a false sense of security by _appearing_ to work, and encouraging you to sink a lot of effort into decorating the file... and _then_ wipe it out when you least expect it.

So I'm afraid this feature is going to stay as-is, at least until-and-unless Squad provides a better out-of-the-box solution for config.
